swag/swag_client.py

```diff
@@ -369,3 +416,294 @@ async def execute_swagdmin_command(self, message):
+            destinataire = message.mentions
+            if len(destinataire) != 1:
+                await message.channel.send(
+                    "Merci de mentionner un destinataire (@Bobby Machin) pour "
+                    "lui donner une partie de cette €agnotte !"
+                )
+                return
+            destinataire = destinataire[0]
+
+            if cagnotte.currency == Currency.SWAG:
+                try:
+                    value = int(
+                        "".join(
+                            argent for argent in splited_command if argent.isnumeric()
+                        )
+                    )
+                except ValueError:
+                    raise InvalidSwagValue
+
+            elif cagnotte.currency == Currency.STYLE:
+                try:
+                    value = Decimal(
+                        "".join(
+                            argent
+                            for argent in splited_command
+                            if argent.replace(".", "").replace(",", "").isnumeric()
+                        )
+                    ).quantize(Decimal(".0001"), rounding=ROUND_DOWN)
+                except ValueError:
+                    raise InvalidStyleValue
+
+            self.swag_bank.receive_from_cagnotte(
+                cagnotte_idx, destinataire.id, value, message.author.id
+            )
+
+            cagnotte_info = cagnotte.get_info()
+            await message.channel.send(
+                "Transaction effectuée avec succès ! \n"
+                "```ini\n"
+                f"[€{cagnotte_idx} {cagnotte_info.name}\t"
+                f"{format_number(value)} {currency_to_str(cagnotte_info.currency)}\t"
+                f"-->\t{destinataire.display_name}]\n"
+                "```"
+            )
+
+            await update_forbes_classement(message.guild, self, self.client)
+
+        elif "partager" in splited_command:
+            cagnotte_idx = get_cagnotte_idx_from_command(splited_command)
+            participants_id = [participant.id for participant in message.mentions]
+
+            (
+                participants_id,
+                gain,
+                gagnant_miette,
+                miette,
+            ) = self.swag_bank.share_cagnotte(
+                cagnotte_idx, participants_id, message.author.id
+            )
+
+            participants_str = []
+            for participant_id in participants_id:
+                user = message.guild.get_member(participant_id)
+                if user == None:
+                    participants_str.append(
+                        await get_guild_member_name(
+                            participant_id, message.guild, self.client
+                        )
+                    )
+                else:
+                    participants_str.append(user.mention)
+
+            participants_mention = ", ".join(participants_str)
+
+            cagnotte_info = self.swag_bank.get_active_cagnotte(cagnotte_idx).get_info()
```

Comment:
A `get_active_cagnotte_info` here would be better I believe

Reply:
Resolved here : https://github.com/BobbySoftwares/Bybbot/pull/7/commits/aee1d9039a3aaf5c16f39c5ed623cac499848e9c